pymongo/topology_description.py

```diff
@@ -1,21 +1,678 @@
+        :param max_election_id: greatest electionId seen from a primary, or None
+        :param topology_settings: a TopologySettings
+        """
+        self._topology_type = topology_type
+        self._replica_set_name = replica_set_name
+        self._server_descriptions = server_descriptions
+        self._max_set_version = max_set_version
+        self._max_election_id = max_election_id
+
+        # The heartbeat_frequency is used in staleness estimates.
+        self._topology_settings = topology_settings
+
+        # Is PyMongo compatible with all servers' wire protocols?
+        self._incompatible_err = None
+        if self._topology_type != TOPOLOGY_TYPE.LoadBalanced:
+            self._init_incompatible_err()
+
+        # Server Discovery And Monitoring Spec: Whenever a client updates the
+        # TopologyDescription from an hello response, it MUST set
+        # TopologyDescription.logicalSessionTimeoutMinutes to the smallest
+        # logicalSessionTimeoutMinutes value among ServerDescriptions of all
+        # data-bearing server types. If any have a null
+        # logicalSessionTimeoutMinutes, then
+        # TopologyDescription.logicalSessionTimeoutMinutes MUST be set to null.
+        readable_servers = self.readable_servers
+        if not readable_servers:
+            self._ls_timeout_minutes = None
+        elif any(s.logical_session_timeout_minutes is None for s in readable_servers):
+            self._ls_timeout_minutes = None
+        else:
+            self._ls_timeout_minutes = min(  # type: ignore[type-var]
+                s.logical_session_timeout_minutes for s in readable_servers
+            )
+
+    def _init_incompatible_err(self) -> None:
+        """Internal compatibility check for non-load balanced topologies."""
+        for s in self._server_descriptions.values():
+            if not s.is_server_type_known:
+                continue
+
+            # s.min/max_wire_version is the server's wire protocol.
+            # MIN/MAX_SUPPORTED_WIRE_VERSION is what PyMongo supports.
+            server_too_new = (
+                # Server too new.
+                s.min_wire_version is not None
+                and s.min_wire_version > common.MAX_SUPPORTED_WIRE_VERSION
+            )
+
+            server_too_old = (
+                # Server too old.
+                s.max_wire_version is not None
+                and s.max_wire_version < common.MIN_SUPPORTED_WIRE_VERSION
+            )
+
+            if server_too_new:
+                self._incompatible_err = (
+                    "Server at %s:%d requires wire version %d, but this "  # type: ignore
+                    "version of PyMongo only supports up to %d."
+                    % (
+                        s.address[0],
+                        s.address[1] or 0,
+                        s.min_wire_version,
+                        common.MAX_SUPPORTED_WIRE_VERSION,
+                    )
+                )
```

Comment:
Would you know if this if statements also needs a break?

Reply:
I don't think it makes a difference in behavior, as if the `if` block is executed the `elif` block won't be.